site/content/docs/development/architecture.md

```diff
@@ -0,0 +1,142 @@
+| [ClusterDelivery](reference/#delivery) | [Deliverable](reference/#deliverable) | [ClusterSourceTemplate](reference/#clustersourcetemplate), [ClusterDeploymentTemplate](reference/#clusterdeploymenttemplate), [ClusterTemplate](reference/#clustertemplate) |
+
+### Templates
+
+Templates create or update resources (i.e. kubectl apply).
+
+Templates consist of:
+* Parameters to pass to `spec.template` or `spec.ytt`
+* The Kubernetes resource yaml as `spec.template` or `spec.ytt` [see Templating](#tbd)
+* **Output paths** which tell Cartographer where to find the output of the Kubernetes resource
+  * The path field depends upon the specific template kind.
+
+![Template](../img/template.jpg)
+
+Templates are typed by the output they produce.
+
+| Output      | Template |
+| ----------- | ----------- |
+| Config | [ClusterConfigTemplate](reference/#clusterconfigtemplate) |
+| Image | [ClusterImageTemplate](reference/#clusterimagetemplate) |
+| Source | [ClusterSourceTemplate](reference/#clustersourcetemplate) |
+| Deployment | [ClusterDeploymentTemplate](reference/#clusterdeploymenttemplate) |
+| | [ClusterTemplate](reference/#clustertemplate) |
+
+### Owners
+
+Owners represent the workload or deliverable, which in many cases refer to a single application's source or image 
+location.
+
+Owners are the developer provided configuration which cause a blueprint to be reconciled into resources.
+
+They consist of:
+* **Labels**: blueprints will select based on the labels of an owner, see [selectors](#selectors) 
+* **Params**: parameters supplied to the blueprint, see [Parameter Hierarchy](#parameter-hierarchy)
+* **Source**: The source reference for the input to the Supply Chain or Delivery Blueprints,
+see [Workload](reference.md/#workload) and [Deliverable](reference.md/#deliverable)
+
+![Owner](../img/owner.jpg)
+
+| Owner      | Blueprint |
+| ----------- | ----------- |
+| Workload | ClusterSupplyChain |
+| Deliverable | ClusterDelivery |
+
+## Theory of operation
+
+Given an owner that matches a blueprint, cartographer reconciles the resources referenced by the blueprint.
+The resources are only created when the inputs are satisfied, and a resource is only updated when it's inputs change.
+This results in a system where a new intrinsic result from one resource can cause other resources to change.
+
+Although Cartographer is not a 'runner of things', a resource can be something as simple as a Job or a CI pipeline.
+However, one advantage of Cartographer's design, is that a resource can also be untriggered. Imagine a Build resource 
+that discovers new base OCI images. If it rebuilds your image, then cartographer will see this new image and update 
+other linked resources.
+
+![Generic Blueprint](../img/generic.jpg)
+<!-- https://miro.com/app/board/uXjVOeb8u5o=/ -->
+
+When Cartographer reconciles an owner, each resource in the matching blueprint is reconciled:
```

Comment:
question:

When you say `each resource in the matching blueprint is reconciled`, does "resource" refer to the template or the kubernetes object? 

Either way, it feels incorrect to say each resource is "reconciled". If it is the kubernetes object, maybe it should be "applied" or "created". If it's the template, perhaps there is a better word we could use. "reconciled" feels like it has too much domain specific meaning that we don't intend in this case.

Reply:
Unfortunately, resource is overloaded. We have a list of resources in the blueprint spec. We changed reconciled to applied.